test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -375,4 +381,22 @@ private String errorMessage(ExecutableSection executableSection, Throwable t) {
     protected boolean randomizeContentType() {
         return true;
     }
+
+    /**
+     * Sniff the cluster for host metadata.
+     */
+    private List<Node> sniffHostMetadata(RestClient client) throws IOException {
+        ElasticsearchNodesSniffer.Scheme scheme =
+            ElasticsearchNodesSniffer.Scheme.valueOf(getProtocol().toUpperCase(Locale.ROOT));
+        /*
+         * We don't want to change the list of nodes that the client communicates with
+         * because that'd just be rude. So instead we find the nodes
+         * returned by the sniffer that correspond with the nodes already in the client
+         * and set the nodes to them. That *shouldn't* change the nodes that the client
+         * communicates with.
```

Comment:
is this comment still valid? Aren't we replacing the nodes with the sniffed ones at this point?

Reply:
You are right! I hadn't removed it.